assets/blocks/quiz/category-question-block/category-question-edit.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useQuestionIndex } from '../question-index';
+import { useQuestionCategories } from '../question-categories';
+import CategoryQuestionSettings from './category-question-settings';
+
+/**
+ * Quiz category question block editor.
+ *
+ * @param {Object}   props
+ * @param {Object}   props.attributes              Block attributes.
+ * @param {Object}   props.attributes.categoryName Category name.
+ * @param {Function} props.setAttributes           Set block attributes.
+ */
+const CategoryQuestionEdit = ( props ) => {
+	const {
+		attributes: {
+			options: { number = 1, category },
+		},
+		clientId,
+	} = props;
+	const index = useQuestionIndex( clientId );
+	const [ , getCategoryTermById ] = useQuestionCategories();
+
+	const nextNumber = index;
+	let range = nextNumber;
+	if ( number !== 1 ) {
+		range += ' - ' + ( nextNumber + number - 1 );
+	}
+
+	const questionIndex = (
+		<h2 className="sensei-lms-question-block__index">{ range }.</h2>
+	);
+
+	const categoryName =
+		getCategoryTermById( category )?.name ?? props.attributes.categoryName;
+
+	return (
+		<>
+			<CategoryQuestionSettings { ...props } />
+			<div
+				className={ `sensei-lms-question-block ${
```

Comment:
I'm sharing this with the question block right now. We could move styles into namespace shared between the two.

Reply:
We can also just think of it as a subtype of the question block and add an extra class to overwrite things.